src/app/js/formats/streamgraph/utils.js

```diff
@@ -177,18 +177,25 @@ export function getMinMaxValue(stackedData) {
     return { minValue, maxValue };
 }
 
-export function cutStr(str) {
-    const strSplit = str.split(' ');
+export function cutStr(str, column_number) {
+    const strSplit = str.split(/-| /);
     let resStr = '';
-    const words_nb = 3;
+    let tokenCounter = 0;
 
-    for (let i = 0; i < strSplit.length && i < words_nb; i++) {
-        if (i < words_nb - 1 || strSplit[i].length > 3) {
-            resStr = `${resStr} ${strSplit[i]}`;
-        }
+    let maxLength;
+    if (column_number == 3) {
+        maxLength = 31;
+    } else maxLength = 31;
```

Comment:
Oh sure it's useless, it was for demonstration purpose, it will be replaced by an extra parameter in this graph to let admins manage it on the fly : 
https://github.com/Inist-CNRS/lodex/blob/cfc410c26805d2b2b5c9c73f1287c6c14b0fb829/src/app/js/formats/streamgraph/Streamgraph.js#L297